src/axom/sina/tests/sina_Document.cpp

```diff
@@ -514,6 +902,221 @@ TEST(Document, saveDocument_hdf5)
+  EXPECT_EQ(ind0_vec, expected_ind0);
+  std::vector<double> ind1_vec = node_to_double_vector(cs0["independent/1/value"]);
+  std::vector<double> expected_ind1 = {7.0, 8.0, 7.0, 9.0};
+  const conduit::Node &csnew = root["records/0/curve_sets/2"];
+  std::vector<double> new_dep0 = node_to_double_vector(csnew["dependent/0/value"]);
+  std::vector<double> expected_new_dep0 = {1.0, 1.0};
+  EXPECT_EQ(new_dep0, expected_new_dep0);
+  std::vector<double> new_dep1 = node_to_double_vector(csnew["dependent/1/value"]);
+  std::vector<double> expected_new_dep1 = {10.0, 10.0};
+  EXPECT_EQ(new_dep1, expected_new_dep1);
+  std::vector<double> new_ind0 = node_to_double_vector(csnew["independent/0/value"]);
+  std::vector<double> expected_new_ind0 = {4.0, 4.0};
+  EXPECT_EQ(new_ind0, expected_new_ind0);
+  std::vector<double> new_ind1 = node_to_double_vector(csnew["independent/1/value"]);
+  std::vector<double> expected_new_ind1 = {7.0, 7.0};
+  EXPECT_EQ(new_ind1, expected_new_ind1);
+
+  // ----- Record 1 -----
+  EXPECT_EQ(root["records/1/type"].as_string(), "foo2");
+  EXPECT_EQ(root["records/1/id"].as_string(), "bar2");
+  const conduit::Node &cs1 = root["records/1/curve_sets/1"];
+
+  std::vector<double> rec1_dep0 = node_to_double_vector(cs1["dependent/0/value"]);
+  std::vector<double> expected_rec1_dep0 = {1.0, 2.0};
+  EXPECT_EQ(rec1_dep0, expected_rec1_dep0);
+  std::vector<double> rec1_dep1 = node_to_double_vector(cs1["dependent/1/value"]);
+  std::vector<double> expected_rec1_dep1 = {10.0, 20.0};
+  EXPECT_EQ(rec1_dep1, expected_rec1_dep1);
+  std::vector<double> rec1_ind0 = node_to_double_vector(cs1["independent/0/value"]);
+  std::vector<double> expected_rec1_ind0 = {4.0, 5.0};
+  EXPECT_EQ(rec1_ind0, expected_rec1_ind0);
+  std::vector<double> rec1_ind1 = node_to_double_vector(cs1["independent/1/value"]);
+  std::vector<double> expected_rec1_ind1 = {7.0, 8.0};
+  EXPECT_EQ(rec1_ind1, expected_rec1_ind1);
+
+  // ----- Record 2 -----
+  EXPECT_EQ(root["records/2/type"].as_string(), "foo2");
+  EXPECT_EQ(root["records/2/id"].as_string(), "bar3");
+
+  const conduit::Node &cs2 = root["records/2/curve_sets/1"];
+  std::vector<double> rec2_dep0 = node_to_double_vector(cs2["dependent/0/value"]);
+  std::vector<double> expected_rec2_dep0 = {1.0, 1.0};
+  EXPECT_EQ(rec2_dep0, expected_rec2_dep0);
+  std::vector<double> rec2_dep1 = node_to_double_vector(cs2["dependent/1/value"]);
+  std::vector<double> expected_rec2_dep1 = {10.0, 10.0};
+  EXPECT_EQ(rec2_dep1, expected_rec2_dep1);
+  std::vector<double> rec2_ind0 = node_to_double_vector(cs2["independent/0/value"]);
+  std::vector<double> expected_rec2_ind0 = {4.0, 4.0};
+  EXPECT_EQ(rec2_ind0, expected_rec2_ind0);
+  std::vector<double> rec2_ind1 = node_to_double_vector(cs2["independent/1/value"]);
+  std::vector<double> expected_rec2_ind1 = {7.0, 7.0};
+  EXPECT_EQ(rec2_ind1, expected_rec2_ind1);
+}
+
+TEST(Document, test_append_to_hdf5_failures)
+{
+  std::streambuf *origBuffer = std::cerr.rdbuf();
```

Comment:
Similarly for this test -- we shouldn't be relying on capturing the output to cerr.